vega_sim/scenario/common/agents.py

```diff
@@ -709,3 +712,206 @@ def finalise(self):
             self.vega.settle_market(
                 self.terminate_wallet_name, self.settlement_price, self.market_id
             )
+
+
+class SemiRandomLimitOrderTrader(StateAgentWithWallet):
+    """Agent which randomly submits and cancels limit orders.
+
+    At initialisation; the agent creates a wallet, identifies the market id and
+    the asset for that market, and mints itself the specified quantity of the
+    required asset.
+
+    At any given step; the agent has an adjustable probability of submitting
+    an order and an adjustable probability of cancelling a randomly
+    selected order.
+
+    When submitting an order; the agent choses a price following a lognormal
+    distribution where the underlying normal distribution can be adjusted.
+
+    """
+
+    def __init__(
+        self,
+        wallet_name: str,
+        wallet_pass: str,
+        market_name: str,
+        asset_name: str,
+        spread: int,
+        initial_asset_mint: float = 1000000,
+        buy_volume: float = 2.5,
+        sell_volume: float = 2.5,
+        tag: str = "",
+        random_state: Optional[np.random.RandomState] = None,
+        submit_bias: float = 0.5,
+        cancel_bias: float = 0.5,
+        side_opts: Optional[dict] = {
+            "SIDE_BUY": 0.5,
+            "SIDE_SELL": 0.5,
+        },
```

Comment:
This should default to `None` then set `foo if side_opts is None else side_opts` in the body of the `__init__`. Will explain why in person.